src/dht.cpp

```diff
@@ -265,31 +265,34 @@ struct Dht::SearchNode {
      * met:
      *
      *  - The node is not expired;
-     *  - If we have heard from the node, we must have heard from him in the last
-     *    NODE_EXPIRE_TIME minutes;
-     *  - The request must not already have been sent;
-     *  - No other request satisfying the request must be pending;
      *  - The pagination process for this particular 'get' must not have begun;
+     *  - There hasn't been any response for a request, satisfying the initial
+     *    request, anytime following the initial request.
+     *  - No other request satisfying the request must be pending;
      *
      * @param now     The time reference to now.
-     * @param update  Time of the last "get" op for the search.
+     * @param update  The time of the last 'get' op satisfying this request.
      * @param q       The query defining the "get" operation we're referring to.
      *
      * @return true if we can send get, else false.
      */
     bool canGet(time_point now, time_point update, std::shared_ptr<Query> q = {}) const {
         /* Find request status for the given query */
-        const auto& get_status = getStatus.find(q);
+        const auto& completed_sq_status = std::find_if(getStatus.cbegin(), getStatus.cend(),
+            [&q,&update](const SyncStatus::value_type& s) {
+                return s.first and q and q->isSatisfiedBy(*s.first) and s.second and s.second->reply_time > update;
```

Comment:
isSatifiedBy is called twice for every getStatus. Maybe there is a way to call it only once ?